### PR TITLE
Add filter item tooltip

### PR DESCRIPTION
**Pull request by qroll**
**Changes**

- Rework popover in form label addon
- Support form label addon in filter item title
- [delete] branch

**Changelog entry**

- New `PopoverV2` component with improved positioning behaviour
- [WARNING] Deprecate `Popover`, to be removed in the next major version

**Additional information**

- Using Floating UI to handle repositioning
  - Encountered build issues as Rollup v2 couldn't handle the library's mjs format, upgraded Rollup to v3
  - Encountered unit test error, updating the `moduleDirectories` resolves it somehow
- Instead of a HOC, use composition with `PopoverTrigger`. Can consider a render prop or hook instead if the composed trigger requires additional data

### CODE REVIEW

**Comment by qroll** [comment]
fixing some alignment issues with the tooltip icon

**Comment by qroll** [comment]
partially addresses https://github.com/LifeSG/react-design-system/issues/373

File: src/form/form-label-addon.tsx [hunk 1]
```diff
@@ -45,83 +11,40 @@ export const AddonTrigger = ({
...
-                setVisible(false);
-            }
+export const PopoverAddon = ({
+    addon,
+    rootNode,
+}: PopoverAddonProps): JSX.Element => {
+    // =========================================================================
+    // CONST, STATE, RE
+    // =========================================================================
+    const { content, type, icon, id, "data-testid": dataTestid } = addon;
```
Comment (keithtxw) [inline src/form/form-label-addon.tsx:24]: could we just name it `testId` for consistency with other components

File: src/popover-v2/popover-trigger.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+export const PopoverTrigger = ({
+    children,
+    popoverContent,
+    trigger = "click",
+    rootNode,
+    onPopoverAppear,
+    onPopoverDismiss,
+    ...otherProps
+}: PopoverV2TriggerProps) => {
+    const [visible, setVisible] = useState<boolean>(false);
```
Comment (keithtxw) [inline src/popover-v2/popover-trigger.tsx:25]: missing the section headers

**Review by keithtxw — CHANGES_REQUESTED** [review]

**Review by keithtxw — APPROVED** [review]

**Pull request merged**